src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +110,216 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        m_rebootThread = std::thread(
+            [scanOrchestrator,
+             &stopRetryThread = m_stopRetryThread,
+             &shouldClean = m_shouldClean,
+             &shouldStop = m_shouldStop,
+             &rebootMutex = m_rebootMutex]()
+            {
+                while (!stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    if (shouldClean.load())
+                    {
+                        try
+                        {
+                            logDebug1(WM_VULNSCAN_LOGTAG, "Perform re-scan after reboot");
+                            std::variant<const SyscollectorDeltas::Delta*,
+                                         const SyscollectorSynchronization::SyncMsg*,
+                                         std::shared_ptr<nlohmann::json>>
+                                data = std::make_shared<nlohmann::json>(nlohmann::json {{"action", "reboot"}});
```

Comment:
This triggers a `ScannerType::PackageRefreshScan` scan type, but this orchestration doesn't remove anything

```
case ScannerType::PackageRefreshScan:
                orchestration = std::make_shared<TFetchFromGlobalDB>();
                orchestration->setNext(std::make_shared<TQueryAllPkgForAgent>(subOrchestration));
                break;
```

Reply:
Fix it!